pkg/query/storeset.go

```diff
@@ -68,6 +69,14 @@ type ExemplarSpec interface {
 	Addr() string
 }
 
+type InfoSpec interface {
+	// Addr returns InofAPI Address for the info spec. It is used as its ID.
+	Addr() string
+	// Metadata returns current labels, min/max ranges for StoreAPI and ExemplarAPI
```

Comment:
Let's start with the smaller abstraction and add this later if we see we need to generalize it to all Info types only if they all need it really. Otherwise, we can keep it in the StoreSpec